src/Reflection/ReflectionParameter.php

```diff
@@ -442,6 +442,10 @@ public function getClass(): ReflectionClass|null
 
         if ($type instanceof ReflectionUnionType) {
             foreach ($type->getTypes() as $innerType) {
+                if (! $innerType instanceof ReflectionNamedType) {
+                    continue;
+                }
```

Comment:
Looks quite broken: perhaps we should drop this method completely from `src/Reflection`, and only keep it on adapters? :thinking:

Reply:
It's good idea, the method is marked as deprecated in PHP. However I prefer to do it in separate PR because the method is used in some unit tests.